pkg/gadgets/top/tcp/tracer/tracer.go

```diff
@@ -203,29 +172,29 @@ func (t *Tracer) nextStats() ([]*types.Stats, error) {
 	}
 
 	for {
-		val := C.struct_traffic_t{}
+		val := tcptopTrafficT{}
 		if err := ips.Lookup(key, unsafe.Pointer(&val)); err != nil {
 			return nil, err
 		}
 
-		srcAddr := C.src_addr(&key)
-		dstAddr := C.dst_addr(&key)
-
 		stat := types.Stats{
-			Saddr:     C.GoString(srcAddr),
-			Daddr:     C.GoString(dstAddr),
-			MountNsID: uint64(key.mntnsid),
-			Pid:       int32(key.pid),
-			Comm:      C.GoString(&key.name[0]),
-			Sport:     uint16(key.lport),
-			Dport:     uint16(key.dport),
-			Family:    uint16(key.family),
-			Sent:      uint64(val.sent),
-			Received:  uint64(val.received),
+			MountNsID: key.Mntnsid,
+			Pid:       int32(key.Pid),
+			Comm:      gadgets.FromCString(key.Name[:]),
+			Sport:     key.Lport,
+			Dport:     key.Dport,
+			Family:    key.Family,
+			Sent:      val.Sent,
+			Received:  val.Received,
 		}
 
-		C.free(unsafe.Pointer(srcAddr))
-		C.free(unsafe.Pointer(dstAddr))
+		if key.Family == syscall.AF_INET {
+			stat.Saddr = gadgets.IPStringFromBytes(key.Saddr, 4)
+			stat.Daddr = gadgets.IPStringFromBytes(key.Daddr, 4)
+		} else {
+			stat.Saddr = gadgets.IPStringFromBytes(key.Saddr, 6)
+			stat.Daddr = gadgets.IPStringFromBytes(key.Daddr, 6)
+		}
```

Comment:
How about using `else if key.Family == syscall.AF_INET6`, then? It would still repeat the 4/6 but would also "catch" the case where family is neither AF_INET nor AF_INET by not filling out the IP address at all.

Reply:
> The thing I am not sure is if we should throw an error if Family is none of AF_INET or AF_INET6.

I'd rather skip those cases inside the ebpf program directly. Right now it wouldn't handle those cases properly as well: see https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/gadgets/top/tcp/tracer/bpf/tcptop.bpf.c#L67

And I don't think we would need to handle miscommunication between bpf and go here, as we check the fields in the integration tests already.

Reply:
Well, there's actually a check: https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/gadgets/top/tcp/tracer/bpf/tcptop.bpf.c#L51

So this should never happen to begin with ;)